File: question2.rb
def solution(riddle)
    alphabet = ('a'..'z').to_a
    n = riddle.length
    result = ''
  
    (0...n).each do |i|
      if riddle[i] == '?'
        prev_char = result[-1]
        next_char = riddle[i + 1]
  
        new_char = alphabet.sample
        new_char = alphabet.sample while new_char == prev_char || new_char == next_char
  
        result += new_char
      else
        result += riddle[i]
      end
    end
  
    result
  end

File: question2.js
function solution(riddle) {
    const alphabet = 'abcdefghijklmnopqrstuvwxyz';
    const n = riddle.length;
    let result = '';

    for (let i = 0; i < n; i++) {
        if (riddle[i] === '?') {
            const prev = result[result.length - 1];
            const next = riddle[i + 1];
            let newChar = alphabet[Math.floor(Math.random() * alphabet.length)];

            while (newChar === prev || newChar === next) {
                newChar = alphabet[Math.floor(Math.random() * alphabet.length)];
            }

            result += newChar;
        } else {
            result += riddle[i];
        }
    }

    return result;
}